src/snowflake/cli/api/entities/common.py

```diff
@@ -1,15 +1,32 @@
 from enum import Enum
+from typing import Generic, Type, TypeVar, get_args
 
 
 class EntityActions(str, Enum):
-    BUNDLE = "bundle"
+    BUNDLE = "action_bundle"
 
 
-class EntityBase:
+T = TypeVar("T")
+
+
+class EntityBase(Generic[T]):
     """
     Base class for the fully-featured entity classes.
     """
 
+    def __init__(self, entity_model: T):
+        self._entity_model = entity_model
+
+    @classmethod
+    def get_entity_model_type(cls) -> Type[T]:
+        """
+        Returns the generic model class specified in each entity class.
+
+        For example, the ApplicationEntityModel class will be returned:
+        ApplicationEntity(EntityBase[ApplicationEntityModel]
```

Comment:
The wording is super unclear to me. Suggestion: 

"""
For example, calling `ApplicationEntity.get_entity_model_type()` returns `ApplicationEntityModel`
"""

or something like that.

Reply:
Done